app/Http/Controllers/Api/AirlineController.php

```diff
@@ -10,30 +10,36 @@
 
 class AirlineController extends Controller
 {
-    public function index()
+    public function index(Request $request)
     {
+        $sortBy = $request->input('sort_by', 'id');
+        $ascending = $request->boolean('asc', true);
+
         return Airline::with('cities')
             ->withCount('flights')
+            ->order($sortBy, $ascending)
+            ->filterByCity($request->input('city'))
+            ->filterByFlights($request->input('flights'))
```

Comment:
You could do the same as you did for the sort in the city and flights, just in case you need to validate something else in the future. Also, there is a validation you can use that makes sure the values exists in the database (https://laravel.com/docs/10.x/validation#rule-exists) that way you can validate that the city id does indeed exists (and you can add sometimes so it wont fail when you don't pass the value)

Reply:
Great. So that would be: (i) adding variables (like `cityId` or `airlineId`) for the inputs I use, and (ii) creating a new Form request for handling the validation of query parameters in the request. Correct?

Reply:
Yes, you could not create a new FormRequest, and just check the validation in the controller (https://laravel.com/docs/10.x/validation#quick-writing-the-validation-logic), although it looks cleaner if you create new FormRequests, but do whatever you want, but al least validate those fields